AutoCollection/HeartBeat.ts

```diff
@@ -61,16 +73,88 @@ class HeartBeat {
             properties["appSrv_wsHost"] = process.env.WEBSITE_HOSTNAME || "";
         } else if (process.env.FUNCTIONS_WORKER_RUNTIME) { // Function apps
             properties["azfunction_appId"] = process.env.WEBSITE_HOSTNAME;
-        } 
-        this._client.trackMetric({name: Constants.HeartBeatMetricName, value: 0, properties: properties});
-        callback();
+        } else if (config) {
```

Comment:
I'm not too familiar on how config is defined. Will this ever be undefined?

Reply:
The config param is only used when checking if it's vm and making http request, because the first param for Util.makeRequest is config. So I pass config as an optional param. For web app and function app it's not necessary.

Reply:
I only see `config` being used `Util.makeRequest`. What is actually being used in `config` that `makeRequest` uses? I do not believe we have any user configurable things so why have a config at all?

Reply:
If a user has specific proxy/agent security requirements for their services, we must honor those for all http(s) requests made by this SDK.